crates/trie/parallel/src/proof.rs

```diff
@@ -317,6 +334,23 @@ where
 
         Ok(MultiProof { account_subtree, branch_node_hash_masks, branch_node_tree_masks, storages })
     }
+
+    /// Returns a [`DecodedMultiProof`] for the given proof.
+    ///
+    /// Uses `multiproof` first to get the proof, and then decodes the nodes of the multiproof.
+    pub fn decoded_multiproof(
+        self,
+        targets: MultiProofTargets,
+    ) -> Result<DecodedMultiProof, ParallelStateRootError> {
+        let multiproof = self.multiproof(targets)?;
+
+        // Now decode the nodes of the multiproof
+        let multiproof = multiproof
+            .try_into()
+            .map_err(|err: alloy_rlp::Error| ParallelStateRootError::Other(err.to_string()))?;
```

Comment:
same re error

Reply:
added `From<alloy_rlp::Error> for ParallelStateRootError`, so this is now just a `?`